kobo/hub/models.py

```diff
@@ -606,9 +606,8 @@ def __str__(self):
         return u"#%s [method: %s, state: %s, worker: %s]" % (self.id, self.method, self.get_state_display(), self.worker)
 
     def save(self, *args, **kwargs):
-        # save to db to precalculate subtask counts and obtain an ID (on insert) for stdout and traceback
```

Comment:
This should be an improvement, but this code *probably* should have used transaction.atomic (like some other of the Task methods do) due to the later call to `self.parent.save()`.  It seems like transaction.atomic is the only way to make sure a task and its parent would both be updated atomically.

Reply:
Addressed in da941ba59656c72e4282e6b8b2a4e90941335e09.